tests/CMakeLists.txt

```diff
@@ -26,7 +26,9 @@ set(SOURCES
 add_executable(tests ${SOURCES} ${HEADERS})
 target_compile_definitions(tests PRIVATE ${PLATFORM_DEFINES})
 target_link_libraries(tests realm-object-store ${PLATFORM_LIBRARIES})
+if(REALM_ENABLE_SYNC)
 target_link_libraries(tests realm-sync)
```

Comment:
I don't think this `target_link_libraries` is necessary. `realm-object-store` lists `realm-sync` as one of its public dependencies, which means that anyone that links to the `realm-object-store` target will automatically pull in `realm-sync`.